### PR TITLE
[KT] Extend one workgroup implementation to sort signed integers & floats

### PR DESCRIPTION
**Pull request by akukanov**
Key changes: Use the sort identity for filling in extra vector lanes, and fix the scatter mask when writing the results back. Also there are some cosmetic improvements, and the test is updated.

### CODE REVIEW

**Review by SergeyKopienko — APPROVED** [review]
LGTM

File: include/oneapi/dpl/experimental/kt/esimd_radix_sort_one_wg.h [hunk 1]
```diff
@@ -53,8 +54,8 @@ void one_wg_kernel(sycl::nd_item<1> idx, uint32_t n, uint32_t THREAD_PER_TG, con
     uint32_t slm_bin_hist_start = 0;
     uint32_t slm_incoming_offset = slm_bin_hist_start + BIN_HIST_SLM_SIZE;
 
-    uint32_t slm_reorder = slm_reorder_start + local_tid * PROCESS_SIZE * sizeof(KeyT);
-    uint32_t slm_bin_hist_this_thread = slm_bin_hist_start + local_tid * BIN_COUNT * sizeof(hist_t);
+    uint32_t slm_reorder_this_thread = slm_reorder_start + local_tid * PROCESS_SIZE * sizeof(KeyT);
```
[the remark this answers is not in the document]
Comment (akukanov) [reply include/oneapi/dpl/experimental/kt/esimd_radix_sort_one_wg.h:57]: Technically not.
I think as the implementation progresses, we will gradually clean up or improve these and other cosmetic things.

[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/experimental/kt/esimd_radix_sort_one_wg.h:57]: ok

**Review by SergeyKopienko — APPROVED** [review]
LGTM

**Review by dmitriy-sobolev — APPROVED** [review]

**Pull request merged**